source/dem/particle_wall_dmt_force.cc

```diff
@@ -0,0 +1,508 @@
+      this->effective_surface_energy[i] =
+        particle_surface_energy + wall_surface_energy -
+        std::pow(std::sqrt(particle_surface_energy) -
+                   std::sqrt(wall_surface_energy),
+                 2);
+
+      const double log_coeff_restitution =
+        log(this->effective_coefficient_of_restitution[i]);
+      this->model_parameter_beta[i] =
+        log_coeff_restitution /
+        sqrt((log_coeff_restitution * log_coeff_restitution) + 9.8696);
+
+      this->effective_coefficient_of_friction[i] =
+        2 * particle_friction_coefficient * wall_friction_coefficient /
+        (particle_friction_coefficient + wall_friction_coefficient + DBL_MIN);
+
+      this->effective_coefficient_of_rolling_friction[i] =
+        2 * particle_rolling_friction_coefficient *
+        wall_rolling_friction_coefficient /
+        (particle_rolling_friction_coefficient +
+         wall_rolling_friction_coefficient + DBL_MIN);
+    }
+
+  if (dem_parameters.model_parameters.rolling_resistance_method ==
+      Parameters::Lagrangian::RollingResistanceMethod::no_resistance)
+    {
+      calculate_rolling_resistance_torque =
+        &ParticleWallDMTForce<dim>::no_resistance;
+    }
+  else if (dem_parameters.model_parameters.rolling_resistance_method ==
+           Parameters::Lagrangian::RollingResistanceMethod::constant_resistance)
+    {
+      calculate_rolling_resistance_torque =
+        &ParticleWallDMTForce<dim>::constant_resistance;
+    }
+  else if (dem_parameters.model_parameters.rolling_resistance_method ==
+           Parameters::Lagrangian::RollingResistanceMethod::viscous_resistance)
+    {
+      calculate_rolling_resistance_torque =
+        &ParticleWallDMTForce<dim>::viscous_resistance;
+    }
+  this->calculate_force_torque_on_boundary =
+    dem_parameters.forces_torques.calculate_force_torque;
+  this->center_mass_container = dem_parameters.forces_torques.point_center_mass;
+  this->boundary_index        = boundary_index;
+  this->force_on_walls        = this->initialize();
+  this->torque_on_walls       = this->initialize();
+}
+
+template <int dim>
+void
+ParticleWallDMTForce<dim>::calculate_particle_wall_contact_force(
+  typename DEM::dem_data_structures<dim>::particle_wall_in_contact
+                            &particle_wall_pairs_in_contact,
+  const double               dt,
+  std::vector<Tensor<1, 3>> &torque,
+  std::vector<Tensor<1, 3>> &force)
+{
+  ParticleWallContactForce<dim>::force_on_walls =
+    ParticleWallContactForce<dim>::initialize();
+  ParticleWallContactForce<dim>::torque_on_walls =
+    ParticleWallContactForce<dim>::initialize();
+
+  // Looping over particle_wall_pairs_in_contact, which means looping over all
+  // the active particles with iterator particle_wall_pairs_in_contact_iterator
+  for (auto &&pairs_in_contact_content :
+       particle_wall_pairs_in_contact | boost::adaptors::map_values)
+    {
+      // Now an iterator (particle_wall_contact_information_iterator) on each
```

Comment:
particle_wall_contact_information_iterator doesnt exist anymore

Reply:
The same comment is written is every particle-wall contact model... 

I just remove the comment entirely?

Reply:
Naaah, let's keep that for the p-w contact forces refactoring then!